tests/mechanic/launcher_test.py

```diff
@@ -115,23 +118,60 @@ def wait(self):
         return 0
 
 
+class MockProcess:
+    def __init__(self, pid):
+        self.pid = pid
+        self.killed = False
+
+    def name(self):
+        return "p{pid}".format(pid=self.pid)
+
+    def wait(self, timeout=None):
+        raise psutil.TimeoutExpired(timeout)
+
+    def kill(self):
+        self.killed = True
+
+
+def get_metrics_store(cfg):
+    cfg.add(config.Scope.application, "track", "params", None)
+    cfg.add(config.Scope.application, "system", "env.name", None)
```

Comment:
IMHO such a function should be free of (globally visible) side effects and we should not modify `cfg` here. I'd be ok if we make track params optional  (with a default value of `{}`) in the constructor of `MetricsStore` but I think all call sites must provide `env.name` when calling this function.

Reply:
Normally, I agree completely!  I compromised here for convenience to reduce the number of changed lines necessary when changing from MockMetricsStore to a real one.  I like your suggestions though, so I'll incorporate them.